src/main/java/com/cloudogu/scm/review/pullrequest/dto/PullRequestMapper.java

```diff
@@ -55,35 +57,44 @@
   @Mapping(target = "author", source = "author", qualifiedByName = "mapAuthor")
   public abstract PullRequestDto map(PullRequest pullRequest, @Context Repository repository);
 
+  @Mapping(target = "subscriber", ignore = true)
   @Mapping(target = "reviewer", source = "reviewer", qualifiedByName = "mapReviewerFromDto")
   public abstract PullRequest map(PullRequestDto dto);
 
   @Named("mapReviewerFromDto")
-  Set<String> mapReviewerFromDto(Set<DisplayedUserDto> reviewer) {
+  Map<String, Boolean> mapReviewerFromDto(Set<ReviewerDto> reviewer) {
     if (reviewer.isEmpty()) {
-      return Collections.emptySet();
+      return Collections.emptyMap();
     }
     return reviewer
       .stream()
-      .map(DisplayedUserDto::getId)
+      .map(ReviewerDto::getId)
       .map(userDisplayManager::get)
       .filter(Optional::isPresent)
       .map(Optional::get)
       .map(DisplayUser::getId)
-      .collect(Collectors.toSet());
+      .collect(Collectors.toMap(p->p, p -> false));
   }
 
   @Named("mapReviewer")
-  Set<DisplayedUserDto> mapReviewer(Set<String> reviewer) {
+  Set<ReviewerDto> mapReviewer(Map<String, Boolean> reviewer) {
     return reviewer
+      .entrySet()
       .stream()
-      .map(userDisplayManager::get)
-      .filter(Optional::isPresent)
-      .map(Optional::get)
-      .map(this::createDisplayedUserDto)
+      .map(entry -> this.createReviewerDto(getUserIfAvailable(entry), entry.getValue()))
       .collect(Collectors.toSet());
   }
 
+  private DisplayUser getUserIfAvailable(Map.Entry<String, Boolean> entry) {
+    Optional<DisplayUser> childDisplayUser = userDisplayManager.get(entry.getKey());
+    DisplayUser user = childDisplayUser.orElse(null);
```

Comment:
What do you mean by childDisplayUser? Naming confuses me.

Reply:
Meaningful naming isn't easy :/